src/interfaces/controllers/insert-trailer-registration/insert-trailer-registration.ts

```diff
@@ -0,0 +1,67 @@
+import { Request, Response, NextFunction } from 'express';
+import * as domain from '../../../domain';
+import { DataAccess } from '../../../utils/data-access';
+import { IRequestHandler } from '../i-request-handler';
+import { insertValidator } from './insert-trailer-validator';
+
+export class InsertTrailerRegistration implements IRequestHandler {
+  private validate(payload?: domain.TrailerRegistration): string {
```

Comment:
I think it is best if we have the validator and DOA as dependency injections of this file for testability or even its separate class. The method calls validate and the only input it takes is the type of validator (insert-trailer vs deregister-trailer)

Reply:
validate is tested via the main call function when we return errors. I need to discuss why DAO needs to be injected.